tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,217 @@
+
+  const needUser = pairPlaceTypesNeedsUsers[type];
+  let parentPersonThirdLevelId = null;
+  let parentPersonSecondLevelId = null;
+  let parentPersonFirstLevelId = null;
+
+  if (type === 'district_hospital') {
+    parentPersonFirstLevelId = place._id;
+  }
+  if (type === 'health_center') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    const randomIndex = Faker.faker.datatype.number(numberOfManagersPerDistrictHospital - 1);
+    place.supervisor = managers[randomIndex]._id;
+  }
+  if (type === 'clinic') {
+    parentPersonFirstLevelId = place._id;
+    parentPersonSecondLevelId = parentFirstLevelId;
+    parentPersonThirdLevelId = parentSecondLevelId;
+  }
+
+  let isPrimaryContact = true;
+  for (let i = 0; i < numberOfPersons; i++) {
+    let roles = pairPlaceTypesRoles[type];
+    let subtype = pairPlaceTypesPersonSubtype[type];
+    if (type === 'clinic' && !isPrimaryContact) {
+      subtype = 'other';
+    }
+    const person = bracPersonFactory.generateBracPerson(setParents(parentPersonFirstLevelId,
+      parentPersonSecondLevelId, parentPersonThirdLevelId), subtype);
+    if (isPrimaryContact) {
+      place.contact = person;
+      createDataDoc(preconditionDataDirectory + place._id + dataExtension, JSON.stringify(place, {}, 2));
+      if (type === 'district_hospital') {
+        roles = ['national_admin', 'mm-online'];
+      }
+    }
+    createDataDoc(preconditionDataDirectory + person._id + dataExtension, JSON.stringify(person, {}, 2));
+    if (type === 'district_hospital') {
+      managers.push(person);
+    }
+    if (needUser) {
+      const personUser = bracUserFactory.generateBracUser(
+        person.short_name.toLowerCase() + placeName + 'user' + i, roles, place._id);
+      createDataDoc(preconditionDataDirectory + personUser._id + dataExtension, JSON.stringify(personUser, {}, 2));
+      const user = {
+        username: person.short_name.toLowerCase() + placeName + 'user' + i,
+        password: personUser.password,
+        roles: personUser.roles,
+        contact: person._id,
+        phone: person.phone,
+        place: personUser.facility_id
+      };
+      users.push(user);
+    }
+    //For every even index number creates a report for the precondition script data
+    //For every odd index number creates a report for the main script data
+    if (person.family_member_type === 'member_eligible_woman' || person.family_member_type === 'member_child') {
+      let reportsDirectory = preconditionDataDirectory;
+      if (i % 2 === 0) {
+        reportsDirectory = preconditionDataDirectory;
+      } else {
+        createDataDirectory(mainDataDirectory, directParentPlace.contact._id);
+        reportsDirectory = mainDataDirectory + directParentPlace.contact._id + '/';
+      }
+
+      if (person.family_member_type === 'member_eligible_woman') {
+        if (person.group_other_woman_pregnancy.other_woman_pregnant) {
```

Comment:
Please flatten this and only use one `if` condition. It would also be very useful if `member_eligible_woman` and `member_child` were not used here, and instead we would export something from the brac-person file like `shouldGeneratePregnancySurvey(person)` and `shouldGenerateAssessmentSurvey(person)`.